lib/Network/OAuth2/Server/Store/PostgreSQL.hs

```diff
@@ -103,31 +103,39 @@ instance TokenStore PSQLConnPool where
                 errorM logName $ "Consistency error: multiple tokens found matching " <> show tok
                 return Nothing
 
-    storeRevokeToken (PSQLConnPool pool) user_id token_id = do
+    storeRevokeToken (PSQLConnPool pool) token_id = do
         withResource pool $ \conn -> do
-            debugM logName $ "Revoking token with id " <> show token_id <> " for user " <> show user_id
-            rows <- execute conn "UPDATE tokens SET revoked = NOW() WHERE (token_id = ?) AND (user_id = ?)" (token_id, user_id)
+            debugM logName $ "Revoking token with id " <> show token_id
+            rows <- execute conn "UPDATE tokens SET revoked = NOW() WHERE (token_id = ?)" (Only token_id)
             case rows of
                 1 -> do
-                    debugM logName $ "Revoked token with id " <> show token_id <> " for user " <> show user_id
-                    return True
+                    debugM logName $ "Revoked token with id " <> show token_id
+                    return ()
                 0 -> do
-                    debugM logName $ "Failed to revoke token " <> show token_id <> " for user " <> show user_id
-                    return False
+                    let msg = "Failed to revoke token " <> show token_id 
+                    errorM logName msg
+                    fail msg
                 x -> do
-                    let errMsg = "Consistency error: revoked multiple (" <> show x <> ") tokens " <> show token_id <> " for user " <> show user_id
-                    errorM logName errMsg
-                    fail errMsg
+                    let msg = "Consistency error: revoked multiple (" <> show x <> ") tokens " <> show token_id
+                    errorM logName msg
+                    fail msg
 
-    storeListTokens (PSQLConnPool pool) uid (review pageSize -> size :: Integer) (review page -> p) = do
+    storeListTokens (PSQLConnPool pool) maybe_uid (review pageSize -> size :: Integer) (review page -> p) =
         withResource pool $ \conn -> do
-            debugM logName $ "Listing tokens for " <> show uid
-            putStrLn "a"
-            tokens <- query conn "SELECT token_id, token_type, token, expires, user_id, client_id, scope FROM tokens WHERE (user_id = ?) AND revoked is NULL ORDER BY created LIMIT ? OFFSET ?" (uid, size, (p - 1) * size)
-            debugM logName $ "Counting number of tokens for " <> show uid
-            putStrLn "b"
-            [Only numTokens] <- query conn "SELECT count(*) FROM tokens WHERE (user_id = ?)" (Only uid)
-            return (map (\((Only tid) :. tok) -> (tid, tok)) tokens, numTokens)
+            (toks, n_toks) <- case maybe_uid of
+                Nothing -> do
+                    debugM logName "Listing all tokens"
+                    toks <- query conn "SELECT token_id, token_type, token, expires, user_id, client_id, scope FROM tokens WHERE revoked is NULL ORDER BY created LIMIT ? OFFSET ?" (size, (p - 1) * size)
```

Comment:
nitpick: Have the same base for the two queries and <> in the extra stuff for user_id?